File: LeetCode/java/2357_Make_Array_Zero_by_Subtracting_Equal_Amounts.java
import java.util.PriorityQueue;

class Solution {
    public int minimumOperations(int[] nums) {
        PriorityQueue<Integer> lo = new PriorityQueue<>();
        PriorityQueue<Integer> lo2 = new PriorityQueue<>();
        int answer = 0;

        for (int n : nums) {
            lo.add(n);
        }

        while (!lo.isEmpty()) {
            int minNum = lo.poll();

            while (minNum == 0 && !lo.isEmpty()) {
                minNum = lo.poll();
            }

            if (minNum == 0 && lo.isEmpty()) {
                break;
            }

            answer++;

            while (!lo.isEmpty()) {
                lo2.add(lo.poll() - minNum);
            }

            while (!lo2.isEmpty()) {
                lo.add(lo2.poll());
            }
        }

        return answer;
    }
}

File: LeetCode/cpp/2357_Make_Array_Zero_by_Subtracting_Equal_Amounts.cpp
class Solution {
public:
	int minimumOperations(vector<int>& nums) {
		priority_queue<int, vector<int>, greater<int>> lo, lo2;
		int answer = 0;

		for (int n : nums) {
			lo.push(n);
		}

		while (!lo.empty()) {
			int min_num = lo.top();
			lo.pop();

			while (min_num == 0 && !lo.empty()) {
				min_num = lo.top();
				lo.pop();
			}

			if (min_num == 0 && lo.empty()) {
				break;
			}

			answer++;

			while (!lo.empty()) {
				lo2.push(lo.top() - min_num);
				lo.pop();
			}

			while (!lo2.empty()) {
				lo.push(lo2.top());
				lo2.pop();
			}
		}

		return answer;
	}
};